Source/ProceduralFairings/FairingBase.cs

```diff
@@ -491,10 +505,13 @@ public void UpdateFairingSideDragCubes()
                 foreach (AttachNode sn in attached)
                 {
                     if (sn.attachedPart is Part sp &&
-                        sp.GetComponent<ProceduralFairingSide>() is ProceduralFairingSide sf &&
-                        sf.dragCubeUpdater is DragCubeUpdater)
+                        sp.GetComponent<ProceduralFairingSide>() is ProceduralFairingSide sf)
                     {
-                        sf.dragCubeUpdater.Update();
+                        sides.Add(sf);
+                        Debug.Log($"sides: {sides.ToString()}");
+                        
+                        if (sf.dragCubeUpdater is DragCubeUpdater)
+                            sf.dragCubeUpdater.Update();
```

Comment:
Don't do this here.  It doesn't clear the sides list, so every time this gets called it keeps appending all the sides that are attached.  But you really don't need to discover the sides here, anyway.  I'd undo this piece, remove the List<ProceduralFairingSide> declaration, and implement below.